### ISSUE TITLE
Style bug: Need to fine-tune the focus style on Button Group (No space)

### ISSUE THREAD

**Issue by orion-cengage** [cengage/react-magma#1293]
The default focus style creates some undesirable visual properties when a button group has no space between buttons. Primarily this is a result of the focus outline overlapping the surrounding buttons. This should be resolvable with some minor tweaks to the focus style and how the buttons are divided, but only on this component.

Screenshot showing visual issues:

![image](https://github.com/cengage/react-magma/assets/24397282/76e4eec5-794d-43ae-89c0-da611c8b8630)

Screenshot of desired result:
